src/layers/base/CanvasLayer.ts

```diff
@@ -9,26 +9,37 @@ export abstract class CanvasLayer extends Layer {
 
   onOpacityChanged(opacity: number): void {
     if (this.canvas) {
-      this.setStyle(this.order, opacity, this.isVisible);
+      this.setStyle();
     }
   }
 
   onOrderChanged(order: number): void {
     if (this.canvas) {
-      this.setStyle(order, this.opacity, this.isVisible);
+      this.setStyle();
+    }
+  }
+
+  onInteractivityChanged(interactive: boolean): void {
+    if (this.canvas) {
+      this.setStyle();
     }
   }
 
   setVisibility(visible: boolean): void {
     super.setVisibility(visible);
     if (this.canvas) {
-      this.setStyle(this.order, this.opacity, visible);
+      this.setStyle(visible);
     }
   }
 
-  setStyle(opacity: number, order: number, visible: boolean = true): void {
-    const visibility = visible ? 'visible' : 'hidden';
-    this.canvas.setAttribute('style', `position:absolute;z-index:${this.order};opacity:${this.opacity};visibility:${visibility}`);
+  setStyle(visible?: boolean): void {
```

Comment:
If we can pass in visible here, maybe we need to set isVisible as well or does isVisible getter use the style?
If we can call setStyle without parameters, maybe it should be called updateStyle() or something instead? A setter without parameters seems a bit odd.

Reply:
good point, i'll change it to be named updateStyle()